django_modern_rest/serialization.py

```diff
@@ -124,3 +126,177 @@ def optimize_endpoint(cls, metadata: 'EndpointMetadata') -> None:
             metadata: Endpoint metadata to optimize.
 
         """
+
+
+@dataclass(slots=True, frozen=True)
```

Comment:
```suggestion
@final
@dataclass(slots=True, frozen=True, kw_only=True)
```

it is not meant to be subclassed.